File: cipher/src/main/webapp/cipher/cipher.js
var Abase = "A".charCodeAt(0);

function encycrptChar(charA, charB) {
  let num1 = charA.charCodeAt(0) - Abase;
  let num2 = charB.charCodeAt(0) - Abase;
  return (num1 + num2)%26;
}

function encycrpt(message) {
  let mChar = message.split("");
  let encycrpted = new Array(0);
  let ci = 0;
  mChar.forEach(function(c) {
    if (c === " ") {
      encycrpted.push(c);
      return;
    }
    let num = encycrptChar(c.toUpperCase(), key[ci]);
    let push = String.fromCharCode(num + Abase);
    encycrpted.push(push)
    ci++;
    if (key[ci] == undefined) ci=0;
  })
  return encycrpted.join("");
}

File: cipher/src/main/java/core/cipher.java
package core;

import java.util.ArrayList;

public class cipher {
	
	private final int Abase = "A".charAt(0);
	private String message;
	private String[] key;
	
	public cipher(String message, String[] key) {
		this.message = message;
		this.key = key;
	}
	
	public String encycrpt() {
		String[] mChar = message.split("");
		ArrayList<String> encycrpted = new ArrayList<String>();
		int cIndex = 0;
		for (String c : mChar) {
			if (c == " ") {
				encycrpted.add(c);
				continue;
			}
			int num = encycrptChar(c.toUpperCase(), key[cIndex]);
			String push = String.valueOf(num + Abase);
			encycrpted.add(push);
			cIndex++;
			if (cIndex >= key.length) cIndex=0;
		}
		StringBuilder sb = new StringBuilder();
		for (String s : encycrpted) {
			sb.append(s);
		}
		return sb.toString();
		
	}
	
	private int encycrptChar (String charA, String charB) {
		int num1 = charA.charAt(0) - Abase;
		int num2 = charB.charAt(0) - Abase;
		return (num1 + num2) % 26;
	}
}
